lib/connection/pool.js

```diff
@@ -158,6 +158,25 @@ var Pool = function(topology, options) {
   this.numberOfConsecutiveTimeouts = 0;
   // Current pool Index
   this.connectionIndex = 0;
+
+  // event handlers
+  const self = this;
+  this._messageHandler = messageHandler(this);
+  this._connectionCloseHandler = function(err) {
+    connectionFailureHandler(self, 'close', err, this);
```

Comment:
I'm worried about getting confused about this in the future. Could we maybe do something along the lines of:

```js
const pool = this;
...
this._connectionCloseHandler = function(err) {
  const connection = this;
  connectionFailureHandler(pool, 'close', err, connection);
}
```

or is that overkill?

Reply:
sure